edx_lint/pylint/filters_docstring/filters_docstring_check.py

```diff
@@ -0,0 +1,117 @@
+"""
+Pylint checker for the format of the docstrings of filters.
+
+A filter's docstring should have the following structure:
+
+1. Description: Any non-empty text followed by a blank line.
+2. Filter Type: A line that starts with "Filter Type:".
+3. Trigger: A line that starts with "Trigger:".
+"""
+
+import re
+
+from pylint.checkers import BaseChecker, utils
+
+from edx_lint.pylint.common import BASE_ID
+
+
+def register_checkers(linter):
+    """
+    Register checkers.
+    """
+    linter.register_checker(FiltersDocstringFormatChecker(linter))
+
+
+class FiltersDocstringFormatChecker(BaseChecker):
+    """Pylint checker for the format of the docstrings of filters."""
+
+    name = "docstring-format-checker"
+
+    DOCSTRING_MISSING_PURPOSE = "filter-docstring-missing-purpose"
+    DOCSTRING_MISSING_TYPE = "filter-docstring-missing-type"
+    DOCSTRING_MISSING_TRIGGER = "filter-docstring-missing-trigger"
+
+    msgs = {
+        ("E%d91" % BASE_ID): (
+            "Filter's (%s) docstring is missing the required purpose section or is badly formatted",
+            DOCSTRING_MISSING_PURPOSE,
+            "filters docstring is missing the required purpose section or is badly formatted",
+        ),
+        ("E%d92" % BASE_ID): (
+            "Filter's (%s) docstring is missing the required type section or is badly formatted",
+            DOCSTRING_MISSING_TYPE,
+            "filters docstring is missing the required type section or is badly formatted",
+        ),
+        ("E%d93" % BASE_ID): (
+            "Filter's (%s) docstring is missing the required trigger section or is badly formatted",
+            DOCSTRING_MISSING_TRIGGER,
+            "filters docstring is missing the required trigger section or is badly formatted",
+        ),
+    }
+
+    options = ()
+
+    @utils.only_required_for_messages(
+        DOCSTRING_MISSING_PURPOSE,
+        DOCSTRING_MISSING_TYPE,
+        DOCSTRING_MISSING_TRIGGER,
+    )
+    def visit_classdef(self, node):
+        """
+        Visit a class definition and check its docstring.
+
+        If the class is a subclass of OpenEdxPublicFilter, check the format of its docstring. Skip the
+        OpenEdxPublicFilter class itself.
+
+        """
+        if not node.is_subtype_of("openedx_filters.tooling.OpenEdxPublicFilter"):
```

Comment:
This checker currently validates both whether the class is a subclass of OpenEdxPublicFilter and whether the class name is exactly OpenEdxPublicFilter but this second validation is redundant 'cause by definition, the OpenEdxPublicFilter class cannot be its own subclass, maybe we could do something like this:

```python
if not node.is_subtype_of("openedx_filters.tooling.OpenEdxPublicFilter") or node.name == "OpenEdxPublicFilter":
    return
```